hoot-core-test/src/test/cpp/hoot/core/scoring/MapComparatorTest.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2012, 2013, 2015, 2016, 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+// Hoot
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/io/OsmXmlReader.h>
+#include <hoot/core/io/OsmXmlWriter.h>
+#include <hoot/core/io/OsmPbfReader.h>
+#include <hoot/core/schema/OsmSchema.h>
+#include <hoot/core/scoring/MapComparator.h>
+#include <hoot/core/visitors/SetTagVisitor.h>
+#include <hoot/core/util/MapProjector.h>
+
+#define int64 opencv_broken_int
+#include <hoot/core/util/OpenCv.h>
+#undef int64
+
+using namespace hoot;
+
+// Tgs
+#include <tgs/Statistics/Random.h>
+
+// Boost
+using namespace boost;
+
+// CPP Unit
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/extensions/TestFactoryRegistry.h>
+#include <cppunit/TestAssert.h>
+#include <cppunit/TestFixture.h>
+
+// Qt
+#include <QDebug>
+#include <QThread>
+
+// Standard
+#include <stdio.h>
+
+class MapComparatorTest : public CppUnit::TestFixture
+{
+  CPPUNIT_TEST_SUITE(MapComparatorTest);
+  CPPUNIT_TEST(runTest);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+
+  void runTest()
+  {
+    OsmXmlReader reader;
+    reader.setAddSourceDateTime(true);
+
+    OsmMap::resetCounters();
+    OsmMapPtr map1(new OsmMap());
+    reader.read("test-files/ToyTestA.osm", map1);
+
+    OsmMap::resetCounters();
+    OsmMapPtr map2(new OsmMap());
+    reader.read("test-files/ToyTestA.osm", map2);
+
+    MapComparator uut;
+
+    // Make sure the maps are the same
+    bool match = uut.isMatch(map1, map2);
+    CPPUNIT_ASSERT(match);
+
+    // Now turn on datetime checking...
+    uut.setUseDateTime();
+
+    // Change dates...
+    SetTagVisitor vtor("source:datetime", "1989-12-13T12:34:56Z");
+    map2->visitRw(vtor);
+
+    // Make sure it fails now!
```

Comment:
I like this test.  Tests both success and then makes sure it fails like it should.  Nicely done.